listener.go

```diff
@@ -27,18 +27,25 @@ func NewListener(l net.Listener) *GracefulListener {
 // A gracefulCon wraps a normal net.Conn and tracks the last known http state.
 type gracefulConn struct {
 	net.Conn
-	lastHTTPState http.ConnState
+	currState http.ConnState
 }
 
 type gracefulAddr struct {
 	net.Addr
-	gconn *gracefulConn
+	gracefulConn *gracefulConn
```

Comment:
honeslty, don't see a much of a gain from this rename, I would actually shorten gconn to conn